SharedPackages/BrowserServicesKit/Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1049,11 +1052,14 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 dnsSettings: dnsSettings,
                 regenerateKey: regenerateKey
             )
-        } catch {
-            if let error = error as? NetworkProtectionError, case .vpnAccessRevoked = error {
-                throw TunnelError.vpnAccessRevoked
+        } catch let error as NetworkProtectionError {
+            switch error {
+            case .vpnAccessRevoked, .noAuthTokenFound:
```

Comment:
The diff looks a bit odd, but the actual change is here.  We now consider `noAuthTokenFound` to be indicative of an a expired subscription for the end of disabling the VPN.

We should ensure that this doesn't disable the VPN for valid users... might be tricky.